### PR TITLE
July 2021 feedback minor edits

### PR DESCRIPTION
**Pull request by alexkb0009**
Contains a number of smaller edits from Trello ticket re: July 2021 feedback, and maybe 1 or 2 tangential fixes (i.e. getGene required a uuid which wasn't in the /embed request for one of the notes -- changed to just utilize `@id` for that and not embed uuid for any of the notes). There are a few more items in the ticket which didn't get to -- either due to lack of time or needing some more clarification -- and will put this PR into draft state if find time/clarity to work on them, otherwise this PR is ready for review.

It might make sense to go through each commit as each commit encapsulates an independent single or subset of features/changes.

### CODE REVIEW

File: src/encoded/static/components/item-pages/CaseView/index.js [hunk 1]
```diff
@@ -743,7 +741,7 @@ const BioinformaticsTab = React.memo(function BioinformaticsTab(props) {
             </div> */}
             <div className="tab-inner-container card">
                 <h4 className="card-header section-header py-3">Quality Control Metrics (QC)</h4>
-                <div className="card-body">
+                <div className="card-body py-0">
```
Comment (alexkb0009) [inline src/encoded/static/components/item-pages/CaseView/index.js:751]: BioInfoStats and BioInfoStatsEntry define their own padding now (`py-3` and `mt-04` + `mb-04`).

File: src/encoded/static/components/item-pages/CaseView/index.js [hunk 2]
```diff
@@ -723,13 +721,13 @@ const BioinformaticsTab = React.memo(function BioinformaticsTab(props) {
...
         <h4 data-family-index={0} className="my-0 d-inline-block w-100">
-            <span className="font-italic text-500">{ familyDisplayTitle }</span>
+            <span className="text-400">{ familyDisplayTitle }</span>
             {/* { pedFileName ? <span className="text-300">{ " (" + pedFileName + ")" }</span> : null } */}
             <button type="button" className="btn btn-sm btn-primary pull-right"
                 data-tip="Click to view the provenance graph for the most up-to-date annotated VCF"
                 onClick={onClick} disabled={(!vcfAtId)}>
                 <i className="icon icon-fw icon-sitemap icon-rotate-90 fas mr-1 small" />
-                View <span className="text-500">Provenance Graph</span>
+                View <span className="text-600">Provenance Graph</span>
```
Comment (alexkb0009) [inline src/encoded/static/components/item-pages/CaseView/index.js:737]: We don't have 'text-500' (or font-weight 500 in general) for the Source Sans webfont we're using (it renders as 400 instead) so adjusted such instances to 400 or 600 where made sense.

File: src/encoded/static/components/browse/columnExtensionMap.js [hunk 3]
```diff
@@ -51,16 +53,16 @@ const MultiLevelColumn = React.memo(function MultiLevelColumn(props){
                 </span>
                 <i className="status-indicator-dot ml-07" data-status={status} data-tip={statusTip || Schemas.Term.toName("status", status)} data-html />
             </div>
-            <h4 className="col-main">
+            <h4 className="col-main" data-tip={titleTip} data-delay-show={titleTipDelayShow} data-html={tooltipEnableHtml}>
```
Comment (alexkb0009) [inline src/encoded/static/components/browse/columnExtensionMap.js:56]: In current stylesheet-rules, there's 'text-truncate' styling applied to the `<span>` element here, so we need to have tooltip stuff on it or the column (as long as has the proper width (of column)) else tooltip will render in midpoint of (potentially-truncated) link if tooltip attributes applied to the mainTitle `<a href>` or whatever inline element, hence these props added to MultiLevelColumn.

(The `mainTitle` prop now tends to have a hyperlink element as its content/value)

File: src/encoded/static/components/item-pages/VariantSampleView/InterpretationSpaceController.js [hunk 4]
```diff
@@ -164,23 +168,16 @@ export class InterpretationSpaceWrapper extends React.Component {
         return ajax.promise(patchURL, 'PATCH', {}, JSON.stringify(cleanedNoteToPatch));
     }
 
-    getNote(uuid, noteType) {
```
Comment (alexkb0009) [inline src/encoded/static/components/item-pages/VariantSampleView/InterpretationSpaceController.js:167]: Removed this method (and embedding of Note UUIDs in general for InterpretationSpace, which no longer needed as a result) altogether since can just be replaced by `ajax.promise(noteAtID + "?datastore=database", 'GET');` if using the `@id` instead of noteType+uuid.

File: src/encoded/static/components/browse/columnExtensionMap.js [hunk 5]
```diff
@@ -330,9 +333,10 @@ export const columnExtensionMap = {
                 statusTip = "Sample exists, but some field is missing: uuid or bam_sample_id.";
             }
             return (
-                <a href={sampleId} className="adv-block-link">
-                    <MultiLevelColumn {...{ mainTitle, date, status, statusTip }} dateTitle="Collected:" topLeft={<span className="accession">{ accession }</span>} />
-                </a>);
+                <MultiLevelColumn {...{ date, status, statusTip }} dateTitle="Collection Date:"
+                    topLeft={<span className="accession text-muted">{ accession }</span>}
+                    mainTitle={<a href={sampleId} className="adv-block-link">{ capitalizeSentence(specimen_type) }</a>} />
```
Comment (alexkb0009) [inline src/encoded/static/components/browse/columnExtensionMap.js:338]: On 2nd thought, using CSS/`.text-capitalize` might've been better... but very negligible, will change possibly if have time.

File: src/encoded/types/variant.py [hunk 6]
```diff
@@ -66,40 +66,41 @@
 ]
 
 SHARED_VARIANT_SAMPLE_EMBEDS = [
-    "variant_notes.note_text",
```
Comment (alexkb0009) [inline src/encoded/types/variant.py:69]: We're loading all the notes via /embed endpoint now so think we should be fine with this edit...

AnnotationSpace has all the embeds as most/all data is not on a linkTo, except maybe transcript.gene for currently-selected transcript.. but that is loaded via AJAX in UI.

File: src/encoded/static/components/item-pages/VariantSampleView/VariantSampleOverview.js [hunk 7]
```diff
@@ -111,18 +111,37 @@ export class VariantSampleOverview extends React.PureComponent {
...
+        if (interpretationTabFromQuery !== null) {
+            interpretationTab = parseInt(interpretationTabFromQuery);
+            if (isNaN(interpretationTab)) {
+                interpretationTab = null;
+            }
+        }
 
         return (
             <div className="sample-variant-overview sample-variant-annotation-space-body">
-                <InterpretationController {...passProps} interpretationTab={parseInt(interpretationTab) !== isNaN ? parseInt(interpretationTab): null} {...{ showInterpretation, caseSource, setIsSubmitting, isSubmitting, isSubmittingModalOpen, newContext, newVSLoading }}>
```
Comment (alexkb0009) [inline src/encoded/static/components/item-pages/VariantSampleView/VariantSampleOverview.js:123]: Might need to compare !== [NaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN) but am unsure.. have usually used isNaN as func for safety in older browsers so am bit more comfortable w. that.

File: src/encoded/static/components/item-pages/CaseView/FilteringTab.js [hunk 8]
```diff
@@ -1,14 +1,14 @@
...
+import React, { useMemo, useCallback } from 'react';
 import queryString from 'query-string';
-import moment from 'moment';
 
 import { console, ajax, JWT } from '@hms-dbmi-bgm/shared-portal-components/es/components/util';
 import { VirtualHrefController } from '@hms-dbmi-bgm/shared-portal-components/es/components/browse/components/VirtualHrefController';
 
 import { FilteringTableFilterSetUI, FilterSetController, SaveFilterSetButtonController, SaveFilterSetPresetButtonController } from './FilteringTableFilterSetUI';
 import { CaseViewEmbeddedVariantSampleSearchTable } from './CaseViewEmbeddedVariantSampleSearchTable';
+import { Alerts } from '../../../../../../../ext-projects/shared-portal-components/es/components/ui/Alerts';
```
Comment (Bianca-Morris) [inline src/encoded/static/components/item-pages/CaseView/FilteringTab.js:11]: This is causing a Module Not Found Error for me...
```
 ERROR in ./src/encoded/static/components/item-pages/CaseView/FilteringTab.js
    Module not found: Error: Can't resolve '../../../../../../../ext-projects/shared-portal-components/es/components/ui/Alerts' in '/Users/bmorris/projects/cgap-portal/src/encoded/static/components/item-pages/CaseView'
     @ ./src/encoded/static/components/item-pages/CaseView/FilteringTab.js 11:0-108 185:6-12
     @ ./src/encoded/static/components/item-pages/CaseView/index.js
     @ ./src/encoded/static/components/index.js
     @ ./src/encoded/static/libs/react-middleware.js
     @ ./src/encoded/static/server.js
```

Should it be `import { Alerts } from '@hms-dbmi-bgm/shared-portal-components/es/components/ui/Alerts';`?

**Review by Bianca-Morris — CHANGES_REQUESTED** [review]
Code looks good aside from the one big problem with that import in `FilteringTab.js`.

I'm going to take a more thorough look at it on local, though, just to double check on the changes to note saving + a few other things.

**Comment by Bianca-Morris** [comment]
One other comment: https://gyazo.com/6fa6a6282e9fe042a8d45e40f14e21b9 at this point it might make sense to just have the Patient Info, Family Info and Pedigree cards take up 1/3 of the window each. It's just a little off, which looks unintentional even if it's not necessarily. Since so many of the changes are cosmetic, it might make sense to deploy it somewhere and have Shannon & other folks look at it. (I'm also just having a hard time getting my inserts to cooperate with me so i can test notes)

File: src/encoded/static/components/item-pages/CaseView/FilteringTab.js (hunk 8 above)
Comment (alexkb0009) [reply src/encoded/static/components/item-pages/CaseView/FilteringTab.js:11]: Yep, thanks!

File: src/encoded/static/scss/encoded/modules/_item-pages.scss [hunk 9]
```diff
@@ -760,7 +760,10 @@ i.status-indicator-dot {
 	border: 1px solid #efefef;
 	padding: 3px 7px;
     font-size: 0.7em;
-    letter-spacing: 1px;
+    letter-spacing: 0.05em;
```
[the remark this answers is not in the document]
Comment (alexkb0009) [reply src/encoded/static/scss/encoded/modules/_item-pages.scss:763]: Hm yeah. Is fine in VS Code but there's also definitely not been good tabs-vs-spaces consistency in our SCSS files. It'd make sense to find/replace all tabs and convert to 4-spaces (or vice versa) (prly in sep. branch) (don't have much as to which; maybe 4-spaces since is used in rest of code).

File: src/encoded/static/components/item-pages/CaseView/FilteringTab.js [hunk 10]
```diff
@@ -183,10 +183,25 @@ export const FilteringTab = React.memo(function FilteringTab(props) {
         </SaveFilterSetButtonController>
     );
 
+    const onFailInitialFilterSetItemLoad = useCallback(function(){
+        if (session) {
+            // todo add sentry.io call here.
```
[the remark this answers is not in the document]
Comment (alexkb0009) [reply src/encoded/static/components/item-pages/CaseView/FilteringTab.js:188]: Right now sentry is available on 4DN and theoretically can be initialized for CGAP (need to see what sentry project/id/config need to be used though). Going to see how is working for another week or so on 4DN (or until we have time to add it in) and then I think will (a) remove _most_ of the sentry calls from 4DN and (b) only add select/few sentry calls on CGAP where it makes sense to report an error -- will try to follow a soft convention of "would you want an email notification about this error? If yes, add sentry here; if not, then don't".
